### PR TITLE
Add web build support using emscripten

### PR DESCRIPTION
**Pull request by bluesillybeard**
I added web support to this project using emscripten. It probably only works on Linux at the moment, I haven't tested it on any other OS.

The changes in the readme file should make it clear what the changes entail.

Side note, this is my first "real" project with Zig, and as such my experience with it is quite limited.

It's probably worth noting that I am using Zig version 0.11.0, and the project uses of a lot of tricks to make it work.

### CODE REVIEW

**Comment by Not-Nik** [comment]
Hey there, first of all thank you very much for your efforts, it looks great so far. I do have a few notes though:

- Your README says to build like so `zig build webexport --sysroot [path to emsdk]/upstream/emscripten`, but there is no webexport step. Building with `zig build <example name> -Dtarget=wasm32-emscripten --sysroot [path to emsdk]/upstream/emscripten` worked great for me.
- Using that command, building works without the `updateTargetForWeb` shenanigans on my MacBook
- The `webExport` function seems unused?

Additionally I'll add your `getModule` fix separately so the commit history is more clear.

**Comment by Not-Nik** [comment]
Don't worry about the merge conflict, we'll cross that bridge when we come to it

**Comment by bluesillybeard** [comment]
Yeah, I still have some work to do in order to make it work more intuitively. I made it work based on the target for the examples, and webexport for the project but I think it would be best to build based on the target for both scenarios.

**Comment by bluesillybeard** [comment]
the webExport function is used in the project_setup.sh

**Comment by bluesillybeard** [comment]
`updateTargetForWeb` seems to not be required in some examples, but is required in others. The reason it's required is in the zig standard library `std/c.zig` is missing `fd_t` which causes an error.
I've submitted [an issue](https://github.com/ziglang/zig/issues/16776) to the Zig developers so they can fix it in the meantime.

**Comment by Not-Nik** [comment]
Then I suggest we keep the `updateTargetForWeb` until that is fixed in a stable release of Zig.

I will convert this PR into a draft; do convert it back once you feel it is ready for review

**Comment by Not-Nik** [comment]
I'm confused about the entry point. Couldn't we define the main function in Zig as an extern function with the C ABI? We'd still build a library, but not use  C as an intermediary.

File: build.zig [hunk 1]
```diff
@@ -149,19 +168,218 @@ pub fn build(b: *std.Build) void {
...
+    \\//The entry point found in the zig project
+    \\extern int web_emscripten_entry_point();
+    \\//this is the method that emscripten calls automatically when the page loads
+    \\int main(int argc, char** argv)
+    \\{
+    \\    //TODO: possibly pass arguments into zig?
+    \\    int status = web_emscripten_entry_point();
+    \\    //emscripten won't print until a newline is formed, and often people forget the last "\n" of a print.
+    \\    // Missing lines can be really annoying, so to avoid that one last line is printed before the program exits.
+    \\    printf("\n");
```
Comment (Not-Nik) [inline build.zig]: This should be 
```c
fflush(stdout);
fflush(stderr);
```

**Comment by bluesillybeard** [comment]
Try to make the main method extern, and you'll get an error: `return type 'anyerror!void' not allowed in function with calling convention 'C'`

Removing `anyerror!` builds fine, but emscripten doesn't treat it as an entry point since it expects it to return a `c_int`, which the main method isn't allowed to do in zig. I also don't want to rely on the exact type of main that someones project uses. (which the web build currently does, I'll have to find a way around that)

The intermediate C file isn't really a big deal anyway, since it's hidden and it all gets linked together nicely.
I think a bigger concern is the `web_emscripten_entry_point.zig` thing, since adding files to the source is really weird.

**Comment by bluesillybeard** [comment]
I did some more testing an apparently I've been wrong about emscripten's ability to find entry points this entire time. It's able to find zig's main method no matter what the function signature is, and errors even work properly.

I'm not sure if something changed since I last tested, or if I just missed something obvious, but either way I'm glad I found out because it means I can remove a lot of the hackiness here.

File: project_setup.sh [hunk 2]
```diff
@@ -12,15 +12,19 @@ echo "generating project files..."
...
 
-pub fn build(b: *std.Build) void {
+pub fn build(b: *std.Build) !void {
     const target = b.standardTargetOptions(.{});
     const optimize = b.standardOptimizeOption(.{});
-
-    var raylib = rl.getModule(b);
-    var raylib_math = rl.math.getModule(b);
+    //web exports are completely separate, due to the amount of hackery required.
+    if (target.getOsTag()) {
```
Comment (Not-Nik) [inline project_setup.sh]: This seems like it either wouldn't build or causes unexpected bugs

**Review by Not-Nik — CHANGES_REQUESTED** [review]

File: project_setup.sh (hunk 2 above)
Comment (bluesillybeard) [reply project_setup.sh]: I am working on changing that area of the code to be more flexible and sensible, so don't worry about it for now.

File: build.zig (hunk 1 above)
Comment (bluesillybeard) [reply build.zig]: I'm not experienced with C so I wasn't aware of those functions, thanks!

Comment (bluesillybeard) [reply build.zig]: Although that part of the code is gone now, since the C entry point was removed

**Comment by Not-Nik** [comment]
Looks good to me, if this is ready I'll merge it.

**Comment by bluesillybeard** [comment]
I think it's pretty much ready to merge, so go ahead if you think so too.

**Review by Not-Nik — APPROVED** [review]

**Pull request merged**